src/qtt/instrument_drivers/gates.py

```diff
@@ -246,9 +246,22 @@ def get_boundaries(self):
             boundaries[gate] = param.vals.valid_values
         return boundaries
 
+    def restrict_boundaries(self, gate_boundaries):
+        """ Restrict boundaries values that can be set on the gates.
+
+        Args:
+            gate_boundaries (dict): a range of allowed values per parameter.
+        """
+        new_boundaries = self.get_boundaries()
```

Comment:
new_boundaries should be renamed to current_boundaries
to get rid of the usage before assignment in the last line of this method you should check first if number of gate_boundaries > 0, then the new_boundaries in the for-loop is assigned to for sure. Plus you don't reassign when there are no gate_boundaries given.